File: episodes/10-cannot-read-undefined/reproduce.js
// A user object — but this one has no "address" field
const user = { name: "Ada", age: 36 };

console.log("Name:", user.name);

// BUG: we assume every user has an address.
// user.address is undefined, and reading .city OF undefined throws.
console.log("City:", user.address.city);

File: episodes/36-python-attributeerror/reproduce.py
# reproduce.py — a tiny user store
def build_user(name, email):
    # build a user record and hand it back to the caller
    user = {"name": name, "email": email, "active": True}
    # BUG: we built the dict but never returned it -> build_user returns None


def greet(user):
    return "Hello, " + user.get("name")   # greet needs a real user dict


u = build_user("Ada Lovelace", "ada@example.com")   # u is None (nothing was returned)
print(greet(u))                                     # 'NoneType' object has no attribute 'get'
